src/pepr/istio/index.ts

```diff
@@ -15,6 +20,11 @@ When(a.Pod)
   .WithLabel("batch.kubernetes.io/job-name")
   .WithLabel("service.istio.io/canonical-name")
```

Comment:
not sure I understand the q

Reply:
If `istio` changes its `service.istio.io/canonical-name` label, is there a clean way to handle that?

Reply:
In terms of this specific code the actual value of the label doesn't matter for us since it's just a shortcut to watch job pods that are mutated/injected by istio. Istio changing the value of the label wouldn't matter, and if they changed the key we would just have to updated accordingly. Deploy order-wise everything should roll out fine, istio would update first, then pepr. Since this code is exclusively targeting jobs there's (in general) no issues with previous existing pods, etc (but could probably make backwards compatible to handle running jobs during the upgrade).

Could also change this to use the job label from k8s and do the "is this injected" check inside the watch, which would bypass the need for the label entirely, at the expense of watching more things.